CMakeLists.txt

```diff
@@ -34,6 +34,10 @@ option(IREE_BUILD_TESTS "Builds IREE unit tests." ON)
 option(IREE_BUILD_SAMPLES "Builds IREE sample projects." ON)
 option(IREE_BUILD_DEBUGGER "Builds the IREE debugger app." OFF)
 
+# Require python3
+find_package(PythonInterp 3 REQUIRED)
```

Comment:
always? or only when building tensorflow integrations?
(nothing should require python for the runtime, right?)

Reply:
Enabling `IREE_BUILD_COMPILER` triggers `add_subdirectory(third_party/glslang EXCLUDE_FROM_ALL)`, which pulls in https://github.com/KhronosGroup/glslang/blob/1ff0c181bb37b06371e2ffa2810d473c5e01c9b7/CMakeLists.txt#L149-L153:
```
# CMake needs to find the right version of python, right from the beginning,
# otherwise, it will find the wrong version and fail later
if(IS_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/External)
    find_package(PythonInterp 3 REQUIRED)
endif()
```

I see this error:
```
[cmake] CMake Error at C:/Program Files/CMake/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:146 (message):
[cmake]   Could NOT find PythonInterp: Found unsuitable version "2.7.15", but
[cmake]   required is at least "3" (found C:/python_27_amd64/files/python.exe)
```

so if we pull in *any* dependency that uses python, we should make it at least "3" before it finds a lower version. The runtime doesn't need python though, so I could put this behind some combination of options.